Gems/AWSCore/Code/Tests/Framework/ServiceClientJobConfigTest.cpp

```diff
@@ -69,6 +69,12 @@ class ServiceClientJobConfigTest
         return "";
     }
 
+    bool HasResourceType(const AZStd::string& resourceKeyName) const override
+    {
+        AZ_UNUSED(resourceKeyName);
+        return false;
+    }
```

Comment:
should use `[[maybe_unused]]` instead
```suggestion
    bool HasResourceType([[maybe_unused]] const AZStd::string& resourceKeyName) const override
    {
        return false;
    }
```

Reply:
Fixed!
The Gem is using both AZ_UNUSED and [[maybe_unused]]... changing this file to all use [[maybe_unused]]

Reply:
I think `AZ_UNUSED` should be deprecated since all of the compilers we support supports it. We should remove it from the coding standards if its there